components/Auth/Login.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Button,
+  Divider,
+  HStack,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import React, { Dispatch, SetStateAction, useState } from 'react'
+import { Controller, ErrorOption, useForm } from 'react-hook-form'
+import { emailSignIn } from 'utils/firebase'
+import * as yup from 'yup'
+import { GoogleSignInComponent } from '.'
+
+interface LoginProps {
+  setCurrentState: Dispatch<SetStateAction<'sign_up' | 'log_in'>>
+}
+
+interface LoginForm {
+  email: string
+  password: string
+}
+
+const loginFormSchema = yup.object({
+  email: yup.string().required().email(),
+  password: yup.string().required(),
+})
+
+const initialState = {
+  email: '',
+  password: '',
+}
+
+const Login = ({ setCurrentState }: LoginProps): JSX.Element => {
+  const methods = useForm({
+    defaultValues: initialState,
+    resolver: yupResolver(loginFormSchema),
+  })
+
+  const [showPassword, setShowPassword] = useState(false)
+
+  const onSubmit = ({ email, password }: LoginForm) => {
+    emailSignIn(email, password)
+      .then((res) => {
+        // TODO handle different types of possible firebase errors and return types also add a toast
+        if (res.code !== 200) {
+          const error: ErrorOption = {
+            message: res.message.substring(9),
+          }
+          methods.setError('password', error)
+        }
+      })
+      .catch((err) => console.log(err))
```

Comment:
Our convention around promises has been `await` / `async` and not `.then` so we should rewrite this `.then` / `.catch`.

Also, any errors returned by `emailSignIn` should be indicated by the status code in its response so catching an error is prob unnecessary?